src/cmd/vm.rs

```diff
@@ -20,136 +27,231 @@ pub struct Args {
     Setup(ArgsSetup),
     Start(ArgsStart),
-    Push(ArgsPush),
 }
 
 #[tracing::instrument(level = "trace")]
 pub fn run(args: &Args) -> Result<()> {
     let config = Config::read()?;
     if !config.is_internal() {
-        return Err(anyhow!(
-            "vm subcommand is currently only supported for google internal use"
-        ));
+        bail!("vm subcommand is currently only supported for google internal use");
     }
 
     match &args.nested {
-        SubCommand::Connect(args) => run_connect(args),
         SubCommand::Setup(args) => run_setup(args),
         SubCommand::Start(args) => run_start(args),
-        SubCommand::Push(args) => run_push(args),
     }
 }
 
 #[derive(FromArgs, PartialEq, Debug)]
-/// connects to a running betty instance via SSH
-#[argh(subcommand, name = "connect")]
-pub struct ArgsConnect {
-    /// extra arguments
+/// run first time setup, installs necessary dependencies
+#[argh(subcommand, name = "setup")]
+pub struct ArgsSetup {
+    /// path to dir where betty.sh exists. If omitted, current directory will be
+    /// used.
+    #[argh(option)]
+    arc: Option<String>,
+
+    /// extra arguments to pass to betty.sh. You can pass other options like
+    /// --extra-args "options".
     #[argh(option)]
-    extra_args: Vec<String>,
+    extra_args: Option<String>,
 }
 
-fn run_connect(_args: &ArgsConnect) -> Result<()> {
+fn run_setup(args: &ArgsSetup) -> Result<()> {
+    let dir = find_betty_script(&args.arc)?;
+
+    println!("Updating packages...");
+    let update_package = Command::new("sudo")
+        .args(["apt", "update"])
+        .spawn()
+        .context("Failed to execute sudo apt update")?;
+    update_package
+        .wait_with_output()
+        .context("Failed to wait for updating packages")?;
+
+    println!("Enable KVM...");
+    enable_kvm()?;
+
+    println!("Installing python packages...");
+    let install_python_package = Command::new("sudo")
+        .args(["apt", "install", "python3-pip", "python3-venv"])
+        .spawn()
+        .context("Failed to install python packages")?;
+    install_python_package
+        .wait_with_output()
+        .context("Failed to wait for installing python packages")?;
+
+    println!("Running betty.sh setup...");
+    let arg = match &args.extra_args {
+        Some(a) => a,
+        None => "",
+    };
+    run_betty(&dir, "setup", arg)?;
+
+    println!("Running gcloud auth login...");
+    let gcloud_auth = Command::new("gcloud")
+        .args(["auth", "login"])
+        .spawn()
+        .context("Failed to run gcloud login gcloud")?;
+    gcloud_auth
+        .wait_with_output()
+        .context("Failed to wait for gcloud auth login")?;
+
     Ok(())
 }
 
-#[derive(FromArgs, PartialEq, Debug)]
-/// run first time setup, installs necessary dependencies
-#[argh(subcommand, name = "setup")]
-pub struct ArgsSetup {
-    /// extra arguments
-    #[argh(option)]
-    extra_args: Vec<String>,
-}
+fn enable_kvm() -> Result<()> {
```

Comment:
can you explain why we're doing all these? doesn't betty.sh setup already do it ?

Reply:
I certainly dont remember doing all this manually (but maybe I just forgot)

Reply:
cool so now you're helping the user out even more than just betty.sh :)

good work